### PR TITLE
Samll fixes

### PR DESCRIPTION
**Pull request by radosuav**
This pull request resolves 3 small bugs:
1. Exception when opening the Matplotlib "Figure options" dialog.
2. Exception when there was an attempt to add a vector layer to the layers' table.
3. No immediate updating of the plot when visibility checkbox or line colour when changed.

### CODE REVIEW

**Comment by etiennesky** [comment]
sorry I have been away for a while. I will try to look into this soon, I just need to setup a testing environment

File: tools/plottingtool.py [hunk 1]
```diff
@@ -266,8 +266,8 @@ def changeColor(self,wdg, library, color1, name):					#Action when clicking the
 			temp1 = wdg.plotWdg.figure.get_axes()[0].get_lines()
 			for i in range(len(temp1)):
 				if name == str(temp1[i].get_gid()):
-					temp1[i].set_color((color1.red() / 255.0 , color1.green() / 255.0 , color1.blue() / 255.0 ,  color1.alpha() / 255.0 ))
-					#wdg.plotWdg.figure.get_axes()[0].redraw_in_frame()
+					# avoid this issue https://github.com/matplotlib/matplotlib/issues/1690/
+					temp1[i].set_color('#%02x%02x%02x' % (color1.red(), color1.green(), color1.blue()))
```
Comment (etiennesky) [inline tools/plottingtool.py]: what is the problem exactly? I am not familiar with this way of defining colors

File: profileplugin.py [hunk 2]
```diff
@@ -309,8 +309,9 @@ def changePlotLibrary(self, item):
 	def addLayer(self, layer1 = None):
 		if layer1 is None:
 			layer1 = self.iface.activeLayer()
-		self.tableViewTool.addLayer(self.iface, self.mdl, layer1)
-		layer1.dataChanged.connect(self.refreshPlot)
+		if isProfilable(layer1):
```
Comment (etiennesky) [inline profileplugin.py]: let me check what isProfilable() function does first

**Comment by etiennesky** [comment]
can you please explain what is the problem and the fix in "Fix visibility checkbox and colour change"? Ideally a link to a file which illustrates the problem would help

File: tools/plottingtool.py (hunk 1 above)
Comment (radosuav) [inline tools/plottingtool.py]: There was a bug in matplotlib that was fixed long time ago (https://github.com/matplotlib/matplotlib/pull/2017) but is not included in matplotlib 1.3.1 that is delivered in OSGeo4W. So without the above change I get an error popping up when I click on the edit curves button:
screenshot

File: profileplugin.py (hunk 2 above)
Comment (radosuav) [inline profileplugin.py]: Without the check I get following exception when trying to add vector layer to the profile tool:
screenshot

**Comment by radosuav** [comment]
"Fix visibility checkbox and colour change" commit updates the colour of a profile line and it's visibility as soon as that colour and visibility is changed. Currently, new profile has to be drawn for the visibility and colour to be updated.

Without the commit (layer 1 is deselected and still visible and colour of layer 2 is changed to brown but still shown as green):
screenshot

With the commit those changes take effect immediately:
screenshot

**Comment by etiennesky** [comment]
Hi I am no longer maintaining this project, if you would like to contribute please make a MR Ron the newer project page

https://github.com/PANOimagen/profiletool

**Pull request closed without merging**